### PR TITLE
Update Ambassador IR module to respect preserve request ID flow

### PR DESCRIPTION
**Pull request by prakharjoshi**
## Description
This is the follow up PR for the changes incorporated at https://github.com/datawire/ambassador/pull/2522

## Related Issues
https://github.com/datawire/ambassador/issues/2218

## Testing
A few sentences describing what testing you've done, e.g., manual tests, automated tests, deployed in production, etc.

## Tasks That Must Be Done
- [ ] Did you update CHANGELOG.md?
- [ ] Did you add or update tests?
- [ ] Did you update documentation?
- [ ] Were there any special dev tricks you had to use to work on this code efficiently?
  + [ ] Did you add them to DEVELOPING.md?
- [ ] Is this a build change?
  + [ ] If so, did you test on both Mac and Linux?

## Other
* If this is a documentation change for a particular release, please open the pull request against the release branch.

### CODE REVIEW

File: python/ambassador/ir/irambassador.py [hunk 1]
```diff
@@ -113,6 +114,7 @@ def __init__(self, ir: 'IR', aconf: Config,
             use_ambassador_namespace_for_service_resolution=False,
             server_name="envoy",
             debug_mode=False,
+            preserve_external_request_id=False
```
Comment (kflynn) [inline python/ambassador/ir/irambassador.py:117]: I think that this logic is correct, but there's a comma missing after the `False` which makes me suspect that this'll blow up. 🙂 How about fix that and we can try to get this landed?

Comment (prakharjoshi) [reply python/ambassador/ir/irambassador.py]: opps, thanks for pointing it out @kflynn. The changes have been done. Thanks

**Comment by kflynn** [comment]
Ah, the local caches need updating. The easy way to do this (which should be described in `DEVELOPING.md`?) is to get a Kubernetes cluster and run `make push pytest-only`, then once it passes, run `make pytest-gold` and commit the results.

**Comment by eddy-curv** [comment]
Any updates on this?

**Comment by prakharjoshi** [comment]
@kflynn I rebased my changes with upstream. I did not understand about local cache 🤔 ? Seems like tests are still failing. while running `make KAT_RUN_MODE=envoy pytest` fails for me on local.

**Comment by kflynn** [comment]
(Note that the changes for everything having two notices logged is actually a separate bug that I'll be tackling shortly. Sigh.)

**Comment by eddy-curv** [comment]
Nice! hope to see this in 1.5.5 :)

**Comment by kflynn** [comment]
This should definitely be in our next release, yes. 🙂

**Comment by kflynn** [comment]
So I've realized that I was wrong, and this did _not_ make it into 1.5.5 -- mea culpa! It has definitively been accepted for 1.6.0, though, stay tuned. 🙁

**Pull request merged**